src/Reflection/ReflectionClass.php

```diff
@@ -242,7 +242,15 @@ public static function createFromNode(
      */
     public function getShortName() : string
     {
-        return $this->node->name;
+        if (!$this->isAnonymous()) {
+            return $this->node->name;
+        }
+
+        $name = 'class@anonymous';
+        if ($this->getFileName() !== null) {
+            $name .= ' ' . $this->getFileName();
```

Comment:
The PHP source code is here: https://github.com/php/php-src/blob/master/Zend/zend_compile.c#L6247